compiler/lib/stdlib.ml

```diff
@@ -341,7 +341,51 @@ module Int32 = struct
       n
 end
 
-module Int31 = struct
+module type Arith_ops = sig
+  type t
+
+  val neg : t -> t
+
+  val add : t -> t -> t
+
+  val sub : t -> t -> t
+
+  val mul : t -> t -> t
+
+  val div : t -> t -> t
+
+  val rem : t -> t -> t
+
+  val logand : t -> t -> t
+
+  val logor : t -> t -> t
+
+  val logxor : t -> t -> t
+
+  val shift_left : t -> int -> t
+
+  val shift_right : t -> int -> t
+
+  val shift_right_logical : t -> int -> t
+end
+
+module Int31 : sig
+  type t
```

Comment:
Regarding the implementation of `Int31`: the two’s complement implementation seems rather simple to me. IMO the it should ideally come with randomized property-based tests, regardless of the implementation.

Reply:
@OlivierNicole, can you take care of adding tests ?

Reply:
The tests I have in mind would add a dependency to `qcheck` though. Is that OK?

Reply:
We could make the test conditional on qcheck.

Reply:
I’m not sure how to do that.

Reply:
@hhugo I added property-based tests in `compiler/tests-num/test_int31.ml`. They actually caught a bug in my implementation of `Int31.shift_right_logical`.

Reply:
Thanks, I'm now very confused but let's move the discussion elsewhere.